truewiki/storage/git.py

```diff
@@ -73,6 +73,7 @@ def __init__(self, ssh_command=None):
         self._files_added = []
         self._files_changed = []
         self._files_removed = []
+        self._git = self.prepare()
```

Comment:
Sadly, this will result in a code-warning (rightfully). `prepare` is an overload, and you shouldn't call those from an `__init__` (as it might not be doing what you think it does :D. Please just assign `self._git` in the function below, and return that value. That should be fine :)

I remember I had to do something here before to fix the CodeQL warning :D :D

Reply:
Yeah, but no need to set it to `None` first (I think). It will raise an exception anyway when it cannot set it, which is exactly what we want :)
Does need testing ofc, as I am not 100% sure that actually works :P

Reply:
> Yeah, but no need to set it to None first (I think). 

Check my latest commit, I added logic to return a special nonce if the git object is `None` and then skip the check in `web_routes`. Would you prefer it just raise an exception?

Reply:
If `_git` is `None`, something is horribly broken, and we should just die :P No need to recover a partial situation in my opinion :)